packages/editor/src/editor.ts

```diff
@@ -0,0 +1,207 @@
+import * as path from 'path';
+import { once } from 'events';
+import { promises as fs } from 'fs';
+import { Readable } from 'stream';
+import spawn from 'cross-spawn';
+
+import { bson } from '@mongosh/service-provider-core';
+import { signatures, ShellInstanceState, TypeSignature } from '@mongosh/shell-api';
+import { ShellResult } from '@mongosh/shell-evaluator';
+
+import type { MongoshBus } from '@mongosh/types';
+
+const beautify = require('js-beautify').js;
+
+export interface EditorOptions {
+  input: Readable;
+  vscodeDir: string;
+  tmpDir: string;
+  instanceState: ShellInstanceState;
+  makeMultilineJSIntoSingleLine: any;
+  loadExternalCode: any;
+}
+
+export class Editor {
+  _input: Readable;
+  _vscodeDir: string;
+  _tmpDir: string;
+  _instanceState: ShellInstanceState;
+  _makeMultilineJSIntoSingleLine: (code: string) => string;
+  _loadExternalCode: (input: string, filename: string) => Promise<ShellResult>;
+  _content: string;
+  print: (...args: any[]) => Promise<void>;
+
+  constructor({ input, vscodeDir, tmpDir, instanceState, makeMultilineJSIntoSingleLine, loadExternalCode }: EditorOptions) {
+    this._input = input;
+    this._vscodeDir = vscodeDir;
+    this._tmpDir = tmpDir;
+    this._instanceState = instanceState;
+    this._makeMultilineJSIntoSingleLine = makeMultilineJSIntoSingleLine;
+    this._loadExternalCode = loadExternalCode;
+    this._content = '';
```

Comment:
Can we name this something like `_lastContent` to make it a bit more obvious that it’s not the current, but the most recent content?